cmd/kube-proxy/app/server_others.go

```diff
@@ -435,15 +430,15 @@ func getLocalDetector(mode proxyconfigapi.LocalMode, config *proxyconfigapi.Kube
 	return proxyutiliptables.NewNoOpLocalDetector(), nil
 }
 
-func getDualStackLocalDetectorTuple(mode proxyconfigapi.LocalMode, config *proxyconfigapi.KubeProxyConfiguration, ipt [2]utiliptables.Interface, nodeInfo *v1.Node) ([2]proxyutiliptables.LocalTrafficDetector, error) {
+func getDualStackLocalDetectorTuple(mode proxyconfigapi.LocalMode, config *proxyconfigapi.KubeProxyConfiguration, nodeInfo *v1.Node) ([2]proxyutiliptables.LocalTrafficDetector, error) {
 	var localDetectors [2]proxyutiliptables.LocalTrafficDetector
 	var err error
 
-	localDetectors[0], err = getLocalDetector(mode, config, ipt[0], nodeInfo)
+	localDetectors[0], err = getLocalDetector(v1.IPv4Protocol, mode, config, nodeInfo)
```

Comment:
IIUC this makes the ordering of v4,v6 fixed, right?  If so, would it make more sense to return a `map[IPFamily]LocalTrafficDetector` or even a struct with `ipv4` and `ipv6` fields, rather than an obscure array?

Reply:
It was always that way. Yes, it's weird. It will go away in the next PR (because we won't ever keep track of two LocalTrafficDetectors at once any more; we'll create an IPv4 LocalDetector when/if we create an IPv4 proxier, and we'll create an IPv6 LocalDetector when/if we create an IPv6 proxier).